pypeit/spectrographs/keck_nirspec.py

```diff
@@ -6,24 +7,1124 @@
+            # Flats and trace frames are typed together
+            return good_exp & self.lamps(fitstbl, 'dome') & (hatch == '1') 
+        if ftype == 'pinhole':
+            # Don't type pinhole frames
+            return np.zeros(len(fitstbl), dtype=bool)
+        if ftype in ['arc', 'tilt']:
+            # TODO: This is a kludge.  Allow science frames to also be
+            # classified as arcs
+            is_arc = self.lamps(fitstbl, 'arcs') & (hatch == '1') 
+            good_exp[is_arc] = True
+            is_obj = self.lamps(fitstbl, 'off') & (hatch == '0') 
+            good_exp[is_obj] = fitstbl['exptime'].data[is_obj] > 60.0
+            return good_exp & (is_arc | is_obj)
+        msgs.warn('Cannot determine if frames are of type {0}.'.format(ftype))
+        return np.zeros(len(fitstbl), dtype=bool)
+
+    def lamps(self, fitstbl, status):
+        """
+        Check the lamp status.
+
+        Args:
+            fitstbl (`astropy.table.Table`_):
+                The table with the fits header meta data.
+            status (:obj:`str`):
+                The status to check. Can be ``'off'``, ``'arcs'``, or
+                ``'dome'``.
+
+        Returns:
+            `numpy.ndarray`_: A boolean array selecting fits files that meet
+            the selected lamp status.
+
+        Raises:
+            ValueError:
+                Raised if the status is not one of the valid options.
+        """
+        if status == 'off':
+            # Check if all are off
+            lamp_stat = [k for k in fitstbl.keys() if 'lampstat' in k]
+            retarr = np.zeros((len(lamp_stat), len(fitstbl)), dtype=bool)
+            for kk, key in enumerate(lamp_stat):
+                retarr[kk,:] = fitstbl[key] == '0'
+            return np.all(retarr, axis=0)
+        if status == 'arcs':
+            # Check if any arc lamps are on
+            lamp_stat = [ 'lampstat{0:02d}'.format(i) for i in range(1,6) ]
+            retarr = np.zeros((len(lamp_stat), len(fitstbl)))
+            for kk, key in enumerate(lamp_stat):
+                retarr[kk,:] = fitstbl[key] == '1'
+            return np.any(retarr, axis=0)
+        if status == 'dome':
+            return fitstbl['lampstat06'] == '1'
+
+        raise ValueError('No implementation for status = {0}'.format(status))
+
+    def order_platescale(self, order_vec, binning=None):
+        """
+        Return the platescale for each echelle order.
+
+        Note that NIRES has no binning.
+
+        Args:
+            order_vec (`numpy.ndarray`_):
+                The vector providing the order numbers.
+            binning (:obj:`str`, optional):
+                The string defining the spectral and spatial binning. **This
+                is always ignored.**
+
+        Returns:
+            `numpy.ndarray`_: An array with the platescale for each order
+            provided by ``order``.
+        """
+        return np.full(order_vec.size, 0.193)
```

Comment:
No critical, but should we pull this value from the detector par?

Reply:
I've updated it to do so now.